OpenSim/Common/Test/testDataTable.cpp

```diff
@@ -0,0 +1,108 @@
+ * OpenSim is developed at Stanford University and supported by the US        *
+ * National Institutes of Health (U54 GM072970, R24 HD065690) and by DARPA    *
+ * through the Warrior Web program.                                           *
+ *                                                                            *
+ * Copyright (c) 2005-2015 Stanford University and the Authors                *
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+#include <OpenSim/Common/TimeSeriesTable.h>
+
+int main() {
+    using namespace SimTK;
+    using namespace OpenSim;
+
+    // Default construct, add metadata to columns, append rows one at a time.
+
+    ValueArray<std::string> labels{};
+    for(unsigned i = 1; i <= 5; ++i)
+        labels.upd().push_back(SimTK::Value<std::string>{std::to_string(i)});
+
+    ValueArray<unsigned> col_index{};
+    for(unsigned i = 1; i <= 5; ++i)
+        col_index.upd().push_back(SimTK::Value<unsigned>{i});
+
+    DataTable::DependentsMetaData dep_metadata{};
+    dep_metadata.setValueArrayForKey("labels", labels);
+    dep_metadata.setValueArrayForKey("column-index", col_index);
+
+    DataTable::IndependentMetaData ind_metadata{};
+    ind_metadata.setValueForKey("labels", std::string{"0"});
+    ind_metadata.setValueForKey("column-index", unsigned{0});
+
+    TimeSeriesTable table{};
+    table.setDependentsMetaData(dep_metadata);
+    table.setIndependentMetaData(ind_metadata);
+
+    SimTK::RowVector_<double> row{5, double{0}};
+
+    for(unsigned i = 0; i < 5; ++i)
+        table.appendRow(0.00 + 0.25 * i, row + i);
+
+    for(unsigned i = 0; i < 5; ++i)
+        table.updRowAtIndex(i) += 1;
+
+    for(unsigned i = 0; i < 5; ++i)
+        table.updRow(0 + 0.25 * i) -= 1;
+
+    try {
+        table.appendRow(0.5, row);
+    } catch (OpenSim::Exception&) {}
+
+    table.updTableMetaData().setValueForKey("DataRate", 600);
+    table.updTableMetaData().setValueForKey("Filename", 
+                                            std::string{"/path/to/file"});
+
+    // Retrieve added metadata and rows to check.
+    assert(table.getNumRows() == unsigned{5});
```

Comment:
`assert()` is only present in Debug builds so won't get executed in our CI builds.